core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -78,7 +79,7 @@ class KafkaController(val config: KafkaConfig,
 
   // have a separate scheduler for the controller to be able to start and stop independently of the kafka server
   // visible for testing
-  private[controller] val kafkaScheduler = new KafkaScheduler(1)
+  private[controller] val autoLeaderRebalanceScheduler = new KafkaScheduler(1, threadNamePrefix = "auto-preferred-leader-rebalance-scheduler-")
```

Comment:
This was meant to be a scheduler for the controller, not just auto leader balancing. Sounds like you're trying to make it more specific. Is there a benefit?

Reply:
It was introduced two years ago and it seems like we hadn't found another case for it. My goal isn't to make it more specific necessarily but make it clearer as to what it does right now.
I thought that if we found a new use case for the scheduler we could revert back the change, not create a new one.

I don't feel strongly about this anyway, let me know if you think it's better to keep it generic

Reply:
I think there's no harm keeping it generic. It is not hard to inspect its usage. Adding a thread name prefix is a good idea though. Maybe just "controller-scheduler" or something like that?